src/test/java/tests/Tests.java

```diff
@@ -0,0 +1,214 @@
+package tests;
+
+import dto.ItemPair;
+import exceptions.CommonTestException;
+import org.apache.commons.collections4.CollectionUtils;
+import org.junit.jupiter.api.*;
+import parser.JsonParser;
+import parser.NoSuchFileException;
+import shop.Cart;
+import shop.RealItem;
+import shop.VirtualItem;
+import utils.Utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static constants.Constants.FILE_NAME_LIST;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class Tests {
+
+
+    private static JsonParser parser = new JsonParser();
+    private static List<Cart> carts;
+    private static final List<ItemPair> itemPairs = Arrays.asList(
+            new ItemPair(FILE_NAME_LIST.get(0), 1, 1),
+            new ItemPair(FILE_NAME_LIST.get(1), 0, 2),
+            new ItemPair(FILE_NAME_LIST.get(2), 0, 3),
+            new ItemPair(FILE_NAME_LIST.get(3), 4, 0),
+            new ItemPair(FILE_NAME_LIST.get(4), 5, 0),
+            new ItemPair(FILE_NAME_LIST.get(5), 0, 0)
+    );
+
+    @BeforeAll
+    public static void setUp() {
+        carts = Utils.generateCart(itemPairs);
+        for (Cart cart : carts) {
+            parser.writeToFile(cart);
+        }
+    }
+
+    @Test
+    public void filesExist() {
+        for (int i = 0; i < 6; i++) {
+            try {
+                Assertions.assertTrue(Files.list(Paths.get("src/main/resources"))
+                                .filter(Files::isRegularFile)
+                                .map(path -> path.getFileName().toString())
+                                .collect(Collectors.toList())
+                                .contains(FILE_NAME_LIST.get(i) + ".json"),
+                        String.format("File %s is exist", FILE_NAME_LIST.get(i) + ".json"));
+            } catch (IOException e) {
+                throw new CommonTestException("Error to read the file");
+            }
+        }
+    }
+
+    @Disabled("Excessive check")
+    @Test
+    public void filesQuantity() {
+        try {
+            Assertions.assertTrue(Files.list(Paths.get("src/main/resources")).count() >= itemPairs.size());
+        } catch (IOException ex) {
+            throw new CommonTestException("Error to read the file", ex);
+        }
+    }
+
+    @Test
+    public void checkCartItemsSaving(){
+        Cart cart = parser.readFromFile(new File("src/main/resources/" + FILE_NAME_LIST.get(0) + ".json"));
+        Field realItems = null;
+        Field virtualItems = null;
+        try {
+            realItems = Cart.class.getDeclaredField("realItems");
```

Comment:
omg), what for reflection here. your task is to test existing class, and not "hack" it

Reply:
fixed